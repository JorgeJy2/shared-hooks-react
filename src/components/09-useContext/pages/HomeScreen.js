import React, { useContext } from 'react';
import { UserContext } from '../context/UserContext';

import '../../01-useState/counter.css';

export const HomeScreen = () => {

    const {user, setUser} = useContext(UserContext);
    console.log(user);
    const handlerLogin = ()=>{
        setUser({

        });
    }
    return (
        <div className="container">
         <h1>Home</h1>
         <hr/> 
         <pre>
             { JSON.stringify(user,null, 3)}
             </pre>  

             <button onClick={handlerLogin} className="btn btn-danger" >Logout</button>
        </div>
    )
}
