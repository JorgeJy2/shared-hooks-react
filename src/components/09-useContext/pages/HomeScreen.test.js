import { mount } from "enzyme";
import { UserContext } from "../context/UserContext";
import { HomeScreen } from "./HomeScreen";

describe('test home screen', () => {
    const user = {
        name: 'jorge',
        email: 'jorge@gmail.com'
    }
    const wrapper = mount(<UserContext.Provider value={
        {
            user,
            setUser: () => { }
        }}>
        <HomeScreen />
    </UserContext.Provider>);
    test('should snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

});