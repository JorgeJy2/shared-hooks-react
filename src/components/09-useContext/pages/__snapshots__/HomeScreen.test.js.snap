// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home screen should snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
   "name": "jorge",
   "email": "jorge@gmail.com"
}
  </pre>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
